raiden-cli/src/raiden.ts

```diff
@@ -0,0 +1,127 @@
+import { Raiden, RaidenChannel, RaidenChannels, ChannelState } from 'raiden-ts';
+import { LocalStorage } from 'node-localstorage';
+import { flatMap, values, filter } from 'lodash';
+import { Subscription } from 'rxjs';
+
+const DEFAULT_CONFIG = {
+  matrixServer: 'https://raidentransport.test001.env.raiden.network',
+  pfs: 'https://pfs.raidentransport.test001.env.raiden.network',
+  pfsSafetyMargin: 1.1,
+  caps: { noDelivery: true, webRTC: true },
+};
+
+export default class RaidenService {
```

Comment:
This would be my strongest suggestion so far: I'd like to avoid a new service/wrapper. I know it's usually good practice to separate implementation from libs specifics, but since CLIs would also be a reference implementation, I'd rather have the SDK's `Raiden` instance used as close to the "surface" as possible, possibly even directly by the route handlers!
And also to be as stateless possible outside of the `Raiden` instance, so it should provide you with all state/variables/info needed at any time.
This way, we get quickly notified if any change breaks something, have simple/direct/straightforward/local views of what needs to be done to get anything in or out of the SDK (methods, data), and can use these requirements as informative requests for changing the SDK's APIs and improving it (e.g. using this to drive our decisions on #357 and other public API patterns), we'll change whatever needed on the SDK, and maybe this may drive improvements/simplifications in the dApp as well, which already wraps quite a lot of data/method calls on its vuex state/actions/services.

Reply:
I'm totally with you and understand the arguments. My simple point is: I think
it is not possible how it is now and the goal is to provide the API. So this
would require to lengthen the whole API development while adapting the SDK. My
goal was to provide a first compatible solution for the user (like Olli with the
train demo). Afterwards we can tinker the internals without the user
recognizing. 

Accessing the SDKs directly from the routes sounds awesome. Unfortunately it is designed in a way that someone must instantiate it, but also share the instance. This sharing of an instance is the main feature this wrapper provides. Furthermore I would not know where you would do the subscriptions and on what these subscriptors interact with. 
As I said, I share your vision. I just don't know how it should work right now.
:shrug:

Reply:
So, just to leave it here for reference, you could go with the [shared] data as something like an interface in the form:
```ts
export interface Cli {
  raiden: Raiden;
  app: Express;
  server: Server;
  router: Router; // needed?
  log: logging.Logger; // default it to logging root logger
}
```
Then you create a simple const object matching this interface, and this will contain the variables which currently are on global scope. You can pass it to several functions, either as a first, explicit `cli: Cli` parameter and access it like `cli.raiden.start()`, or pass it as `this` parameter, declaring like:
```ts
function shutdown(this: Cli, ...args: any[]) {
  this.raiden.stop(); // this is properly typed
}
```

You can then call it then like `shutdown.apply(cli, other_params_array)`.
If you need to pass this function ahead as a callback to some place which may not bind `this` (e.g. route handler), you can pass it like:
```ts
async function handleChannelOpen(this: Cli, request: Request, _response: Response, next: NextFunction) {
  await this.raiden.openChannel(...)
}
export function setupChannelsApis(this: Cli) {
  this.router.get('/channels', handleChannelOpen.bind(this));
}
```
You can reassign members (e.g. `this.log = logging.getLogger(`cli:${this.raiden.address}`);`) if needed without problems, as long as you always access them as members from the `cli` object reference (instead of passing the members directly somewhere)
If some function needs to be called when some of these instances are not ready yet, you can pass it like `function(this: Pick<Cli, 'app' | 'server'>)`, and this will require a subset of the interface where only `app` and `server` properties are defined, not others possibly undefined members.
Try as much as possible to have a first `setupCli` or whatever entrypoint, so this can be easily used to setup a CLI instance wherever needed, and this would be the entrypoint to setup everything on the `main` function.

Reply:
At last, you could have your implementation directly on the router definitions, using directly the SDK instance (`this.raiden.`...), and take out only shared logic like filtering or object schema mapping/transform. This way, one can reference directly the endpoint as a reference implementation to how to perform some operation, what specific errors need to be handled, etc.

Reply:
Wow thank you so  much that you did not only developed this idea during our team call, but also left this long written explanation here! :pray:

As for the other discussion will close this one here as well. They lead to a major rework that will require a new review afterwards.